File: android app/app/src/main/java/com/example/samved/model/Complaint.kt
package com.example.samved.model

data class Complaint(
    val _id: String,
    val issueType: String,
    val description: String,
    val ward: String,
    val addressText: String,
    val latitude: Double,
    val longitude: Double,
    val beforePhoto: String,
    val afterPhoto: String?,
    val status: String,
    val priorityScore: Int,
    val reporters: List<String>

)

File: samved-backend/models/Complaint.js
const mongoose = require("mongoose");

const complaintSchema = new mongoose.Schema({
  userId: { type: String, required: true },

  issueType: { type: String, required: true },

  description: { type: String, required: true },

  ward: { type: String, required: true },

  latitude: { type: Number, required: true },
  longitude: { type: Number, required: true },

  addressText: { type: String, required: true },

  reporters: {
    type: [String],
    default: []
  },

  priorityScore: {
    type: Number,
    default: 0
  },

  beforePhoto: { type: String, required: true },
  afterPhoto: { type: String },

  status: {
    type: String,
    enum: ["Pending", "In Progress", "Resolved"], // 🔥 FIX
    default: "Pending"
  }

}, { timestamps: true });

module.exports = mongoose.model("Complaint", complaintSchema);
